### PR TITLE
Fix #9892 thumb max size filter restored

### LINKED ISSUE #9892: BG thumbnails size check no longer works
## Description

After what done for https://github.com/geosolutions-it/MapStore2/issues/9867 the size check of uploaded images for BG thumbnails no longer works.

## How to reproduce

- Open a map with edit rights  
- Open the Background selector to edit a BG thumbnail
- Try to upload an image of size greater than 500kb (in both available ways like selecting the file or using drag&drop)

*Expected Result*

MS should warn the user that the file size exceed the limit.

*Current Result*

MS always allow to upload thumbnail even if the size is exceeding the limit.

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

### PR DESCRIPTION
**Pull request by MV88**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

Fix #9892

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/misc/Thumbnail.jsx [hunk 1]
```diff
@@ -89,20 +89,21 @@ const Thumbnail = forwardRef(({
 
     const handleDrop = (files) => {
         const imageType = files?.[0]?.type;
+        const imageSize = files?.[0]?.size;
```
Comment (allyoucanmap) [inline web/client/components/misc/Thumbnail.jsx:92]: I would propose a different approach, a new prop called `checkOriginalFileSize`, if true we can check directly here without performing the resize of the image, something like this:

Note: checkOriginalFileSize should be set true only for the background dialog

```js
const imageSize = files?.[0]?.size;
if (checkOriginalFileSize && imageSize >= maxFileSize) {
    return onError(['SIZE'], files);
}
```

This should restore previous behaviour and it should not affect other thumbnails that are not using the original image size limit.

File: web/client/components/misc/Thumbnail.jsx [hunk 2]
```diff
@@ -89,20 +89,21 @@ const Thumbnail = forwardRef(({
...
                 setPending(false);
-                if (isASupportedImage && data && size < maxFileSize) {
+                if (isASupportedImage && data && imageSize < maxFileSize) {
                     return onUpdate(data, files);
                 }
                 return onError([
                     ...(!isASupportedImage ? ['FORMAT'] : []),
-                    ...(data && size >= maxFileSize ? ['SIZE'] : [])
+                    ...(data && imageSize >= maxFileSize ? ['SIZE'] : [])
                 ], files);
```
Comment (allyoucanmap) [inline web/client/components/misc/Thumbnail.jsx:107]: Based on previous comment this should be restored as before

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/misc/Thumbnail.jsx [hunk 3]
```diff
@@ -45,6 +45,7 @@ const getThumbnail = (files, options) => {
  * @prop {node} message display node message inside thumbnail
  * @prop {string} thumbnail source of thumbnail
  * @prop {number} maxFileSize max size of file
+ * @prop {boolean} checkOriginalFileSize if tru it checks maxFileSize compared to source image
```
Comment (tdipisa) [inline web/client/components/misc/Thumbnail.jsx:48]:
```suggestion
 * @prop {boolean} checkOriginalFileSize if true it checks maxFileSize compared to source image
```

**Review by tdipisa — CHANGES_REQUESTED** [review]

**Review by allyoucanmap — APPROVED** [review]

**Review by tdipisa — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Pull request merged**